tests/network/port_forward.go

```diff
@@ -0,0 +1,143 @@
+	"kubevirt.io/client-go/kubecli"
+	"kubevirt.io/kubevirt/tests"
+	"kubevirt.io/kubevirt/tests/console"
+	"kubevirt.io/kubevirt/tests/libvmi"
+)
+
+var _ = SIGDescribe("[Serial] Port-forward", func() {
+	var (
+		err        error
+		virtClient kubecli.KubevirtClient
+	)
+
+	BeforeEach(func() {
+		virtClient, err = kubecli.GetKubevirtClient()
+		tests.PanicOnError(err)
+
+		tests.BeforeTestCleanup()
+	})
+
+	Context("VMI With masquerade binding", func() {
+		const (
+			localPort       = 1500
+			declaredPort    = 1501
+			nonDeclaredPort = 1502
+		)
+		var portForwardCmd *exec.Cmd
+
+		When("performing port-forward from a local port to a VMI's pod declared port", func() {
+			var (
+				ports = []v1.Port{
+					{Port: declaredPort},
+				}
+				vmiWithDeclaredPort *v1.VirtualMachineInstance
+			)
+
+			BeforeEach(func() {
+				vmiWithDeclaredPort = createCirrosVMIWithPortsAndBlockUntilReady(virtClient, ports)
+				tests.StartHTTPServer(vmiWithDeclaredPort, declaredPort)
+
+				vmiPod := tests.GetRunningPodByVirtualMachineInstance(vmiWithDeclaredPort, tests.NamespaceTestDefault)
+				portForwardCmd, err = portForwardCommand(vmiPod.Name, vmiPod.Namespace, localPort, declaredPort)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(portForwardCmd.Start()).To(Succeed())
+			})
+
+			It("should reach the vmi", func() {
+				defer func() {
+					Expect(killPortForwardCommand(portForwardCmd)).To(Succeed())
+				}()
+
+				By(fmt.Sprintf("checking that service running on port %d can be reached", declaredPort))
+				Eventually(func() error {
+					err = exec.Command("curl", fmt.Sprintf("127.0.0.1:%d", localPort)).Run()
+					return err
+				}, 40*time.Second, 5*time.Second).ShouldNot(HaveOccurred())
+			})
+		})
+
+		When("performing port-forward from a local port to a VMI's pod non-declared port", func() {
+			var vmiWithoutDeclaredPort *v1.VirtualMachineInstance
+
+			BeforeEach(func() {
+				vmiWithoutDeclaredPort = createCirrosVMIWithPortsAndBlockUntilReady(virtClient, []v1.Port{})
+				tests.StartHTTPServer(vmiWithoutDeclaredPort, nonDeclaredPort)
+
+				vmiPod := tests.GetRunningPodByVirtualMachineInstance(vmiWithoutDeclaredPort, tests.NamespaceTestDefault)
+				Expect(vmiPod).ToNot(BeNil())
+				portForwardCmd, err = portForwardCommand(vmiPod.Name, vmiPod.Namespace, localPort, nonDeclaredPort)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(portForwardCmd.Start()).To(Succeed())
+			})
+
+			It("should reach the vmi", func() {
+				defer func() {
+					Expect(killPortForwardCommand(portForwardCmd)).To(Succeed())
+				}()
+
+				By(fmt.Sprintf("checking that service running on port %d can be reached", nonDeclaredPort))
+				Eventually(func() error {
+					err = exec.Command("curl", fmt.Sprintf("127.0.0.1:%d", localPort)).Run()
+					return err
+				}, 40*time.Second, 5*time.Second).ShouldNot(HaveOccurred())
+			})
+		})
+	})
+})
+
+func portForwardCommand(podName string, podNamespace string, sourcePort, targetPort int) (*exec.Cmd, error) {
```

Comment:
What do you think about passing only the pod as the first parameter

Reply:
Sorry @rhrazdil  for the late reply.

I think you are right. no point of passing two strings when I can pass just the object.
changed.